src/Instrumentation/Psr15/README.md

```diff
@@ -0,0 +1,26 @@
+# OpenTelemetry PSR-15 auto-instrumentation
+
+## Requirements
+
+* OpenTelemetry extension
+* OpenTelemetry SDK and exporters (required to actually export traces)
+
+## Overview
+Auto-instrumentation hooks are registered via composer, and spans will automatically be created for each psr-15 middleware that is executed.
+
+To export spans, you will need to create and register a `TracerProvider` early in your application's lifecycle:
+
+```php
+<?php
+require_once 'vendor/autoload.php';
+
+$tracerProvider = <create tracer provider>;
```

Comment:
Do we plan to simplify that in production ready release?. Important aspect of auto-instrumentation is easy of use. Using this approach user needs to know some details of opentelemetry, at least how to create trace provider and export spans, then he has to add opentelemetry dependencies to application. This might be good for experienced user that want to know what's going on under the hood, but 99% users IMHO will be not interested in such details.

Reply:
If all inputs are provided via environment variables, we could automatically create and register everything. Combining https://github.com/open-telemetry/opentelemetry-php/blob/main/examples/traces/features/configuration_from_environment.php with the composer autoload-files technique is one way I think we could provide zero-code instrumentation.

Reply:
That's what I was thinking about

Reply:
With https://github.com/open-telemetry/opentelemetry-php/pull/849 we could define lazy initializers in the SDK that are invoked iff the user does not configure the instances manually.
```php
Globals::registerInitializer(static function(Configurator $configurator): Configurator {
    $tracerProvider = (new TracerProviderFactory('should be fetched from resource instead'))->create();
    ShutdownHandler::register($tracerProvider->shutdown(...));

    return $configurator->withTracerProvider($tracerProvider);
});
```

Reply:
I think `Globals` already takes care of that - eg if you try to get a tracer provider, it will first check in context:
```php
    public static function tracerProvider(): TracerProviderInterface
    {
        return Context::getCurrent()->get(ContextKeys::tracerProvider()) ?? self::globals()->tracerProvider;
    }
```